modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/add_to_cart/AddToCartButton.es.js

```diff
@@ -243,7 +182,68 @@ class AddToCartButton extends Component {
 			return null;
 		}
 
-		doSubmit.call(this);
+		this._doSubmit();
+	}
+
+	_doSubmit() {
+		const formData = new FormData();
+
+		formData.append('commerceAccountId', this.accountId);
+		formData.append('groupId', themeDisplay.getScopeGroupId());
+		formData.append('productId', this.productId);
+		formData.append('languageId', themeDisplay.getLanguageId());
+		formData.append('quantity', this.inputQuantity);
+		formData.append('options', this.options);
+
+		if (this.orderId) {
+			formData.append('orderId', this.orderId);
+		}
+
+		return fetch(this.cartAPI, {
+			body: formData,
+			method: 'POST',
+		})
+			.then((response) => response.json())
+			.then((jsonResponse) => {
+				if (jsonResponse.success) {
+					if (jsonResponse.orderId !== this.orderId) {
+						this.orderId = jsonResponse.orderId;
+					}
+
+					Liferay.fire(CURRENT_ORDER_UPDATED, {
+						id: jsonResponse.orderId,
+						orderStatusInfo: jsonResponse.orderStatusInfo,
+					});
+
+					this._animateMarker(this.quantity);
+					this.quantity = this.inputQuantity;
+					this._resetInputQuantity(this);
+				}
+				else if (jsonResponse.errorMessages) {
+					showNotification(jsonResponse.errorMessages[0], 'danger');
+				}
+				else {
+					const validatorErrors = jsonResponse.validatorErrors;
+
+					if (validatorErrors) {
+						validatorErrors.forEach((validatorError) => {
+							showNotification(validatorError.message, 'danger');
+						});
+					}
+					else {
+						showNotification(jsonResponse.error, 'danger');
+					}
+				}
+			})
+			.catch(console.error);
```

Comment:
I would use `console.error` as a method of last resort because customers can misconstrue it as evidence of bugs when in fact errors can arise for benign reasons (like network connectivity issues). For more details see [the frontend guidelines](https://github.com/liferay/liferay-frontend-guidelines/blob/master/dxp/debugging.md#logging), as well as suggested alternatives.